internal/core/component_factory.rs

```diff
@@ -0,0 +1,161 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+#![warn(missing_docs)]
+
+//! This module contains the `ComponentFactory` that is exposed to the C API
+
+use vtable::*;
+
+use crate::component::ComponentRc;
+
+/// A `ComponentFactory` used can create a `Component`
+#[vtable]
+#[repr(C)]
+pub struct ComponentFactoryVTable {
+    /// Visit the children of the item at index `index`.
+    /// Note that the root item is at index 0, so passing 0 would visit the item under root (the children of root).
+    /// If you want to visit the root item, you need to pass -1 as an index.
+    pub build: extern "C" fn(core::pin::Pin<VRef<ComponentFactoryVTable>>) -> Option<ComponentRc>,
```

Comment:
If one make the ComponentFactory public, you want to return here a ComponentFactoryRc  here.